doc/manage-duniverse.mld

```diff
@@ -0,0 +1,116 @@
+{1 Considerations}
+
+When using opam-monorepo the question arises: how to manage it in comparison to
+the regular opam workflow? Compared to regular opam usage using opam-monorepo
+brings lockfiles and a duniverse folder into the repository. There are multiple
+ways to handle this, with multiple advantages and disadvantages. This document
+explains the considerations behind dealing with each of those, along with our
+recommendations.
+
+{1 Methods}
+
+We assume the opam file is committed to the repository. Running [opam monorepo
+lock] will create a [.locked] opam file in addition to the original opam file.
+Calling [opam monorepo pull] will create a [duniverse] folder into which it
+will unpack the sources of the packages that are mentioned in the lockfile.
+
+{2 Not locked}
+
+While this is not a [opam-monorepo] workflow per se, projects that do not use
+[opam-monorepo] can still be used with [opam-monorepo]. As such every
+contributor is responsible for creating and maintianing their own lockfile and
+duniverse.
+
+Advantages include:
+
+{ul
+{- No opt-in to [opam-monorepo] required, every developer can chose to use it
+   or not}
+}
+
+Unfortunately, in this approach the disadvantages dominate:
+
+{ul
+{- Everyone might end up with different packages in their duniverse, thus low
+   reproducability}
+{- Everyone needs to have opam-monorepo installed}
+}
+
+The [opam-monorepo] maintainers don't recommend this approach, as is has hardly
+any advantages over an [opam-monorepo]-less approach.
+
+{2:lockfile-in-git Lockfile in Git}
+
+Another approach is very similar to the regular opam workflow, but in addition
+to having the [opam] file committed to the repository, also the lockfile that
+[opam monorepo lock] generates is added.
+
+Advantages include:
+
+{ul
+{- The git repostitory does not grow by much, only one file}
+{- The packages to be installed are reproducible}
+{- Installation on the packages regularly into the Opam switch is still
+   possible}
+}
+
+Disadvantages might be
+
+{ul
+{- Everyone needs to have opam-monorepo installed to pull the dependencies}
```

Comment:
I guess you can still use the lock file with opam? Also `opam-monorepo` being a plugin, it could automatically be installed by `opam` when you call it.

Reply:
To complete that list :
1. users have to have opam installed *and* set up (with `opam init -ay`, a proper environment, etc)
2. the more dependencies you have, the more likely a network issue can occur and make `pull` fail.